modules/ding_permissions/ding_permissions.module

```diff
@@ -1011,3 +1013,14 @@ function ding_permissions_secure_permissions($role) {
     return $permissions[$role];
   }
 }
+
+/**
+ * Implements hook_page_alter().
```

Comment:
I don't see it as a big problem that uid=1 has two toolbars. As administrator you can easily just disable one of the modules.

We removed hook_page_alter() in http://platform.dandigbib.org/issues/2184 in favor of handling it in hook_secure_permission. I see no reason to reintroduce this hook just for uid=1. If permissions is set correctly up for all the roles, there shouldn't really be any problem.